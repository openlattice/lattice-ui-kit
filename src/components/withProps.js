import React from 'react';

const withProps = (WrappedComponent, hocProps) => {
  const LatticeWrapper = props => (
    <WrappedComponent {...hocProps} {...props} />
  );
  return LatticeWrapper;
};

export default withProps;
