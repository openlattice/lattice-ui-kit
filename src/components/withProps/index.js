import withProps from './withProps';

export default withProps;
