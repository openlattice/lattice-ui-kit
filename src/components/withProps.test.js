import React, { Component } from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withProps from './withProps';

class WrappedComponent extends Component {
  render() {
    return (
      <div>test</div>
    );
  }
}

const testProps = {
  open: 'lattice',
  foo: 'bar'
};

describe('withProps', () => {
  test('render matches snapshot', () => {
    const ComponentWithProps = withProps(WrappedComponent, testProps);
    const wrapper = shallow(<ComponentWithProps />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('returns LatticeWrapper', () => {
    const ComponentWithProps = withProps(WrappedComponent, testProps);
    const wrapper = mount(<ComponentWithProps />);
    expect(
      wrapper.find('LatticeWrapper').exists()
    ).toBeTruthy();
  });

  test('attaches props to WrappedComponent', () => {
    const ComponentWithProps = withProps(WrappedComponent, testProps);
    const wrapper = shallow(<ComponentWithProps />);
    expect(
      wrapper.props()
    ).toEqual(testProps);
  });

  test('includes outer props', () => {
    const ComponentWithProps = withProps(WrappedComponent, testProps);
    const wrapper = shallow(<ComponentWithProps outer="included" />);
    expect(
      wrapper.prop('outer')
    ).toEqual('included');
  });
});
