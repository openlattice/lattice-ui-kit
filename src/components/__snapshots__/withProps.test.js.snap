// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withProps render matches snapshot 1`] = `
<WrappedComponent
  foo="bar"
  open="lattice"
/>
`;
